docs/docs/adrs/adr-018-permissionless-ics.md

```diff
@@ -0,0 +1,304 @@
+  // maximum number of validators that can validate a consumer chain
+  uint32 validator_set_cap;
+  // allowlist of provider consensus addresses of validators that are the ONLY ones that can validate the consumer chain
+  repeated string allowlist;
+  // denylist of provider consensus addresses of validators that are the CANNOT validate the consumer chain
+  repeated string denylist;
+}
+```
+
+We store the `ConsumerChainRecord`s using the `consumerID` as their key. The `owner_address` of a consumer chain corresponds
+to the address that issued the `MsgLaunchConsumerChain` and the `owner_address` can be updated by issuing a `MsgUpdateConsumerChain` (see later).
+Note that a `ConsumerChainRecord` is only created when a consumer chain moves from the prelaunched to the _launched_ phase.
+The information needed to set up a `ConsumerChainRecord` are stored in the corresponding `ConsumerAdditionProposal` (see `MsgLaunchConsumerChain` below).
+
+### New Messages
+In what follows, we describe the new messages that Permissionless ICS introduces, and on how those can be used.
+We then, describe how we can utilize those messages with our existing codebase.
+
+#### Launch a Consumer Chain
+To prepare a consumer chain for launch, we issue a `MsgLaunchConsumerChain` message that is as follows:
+
+```protobuf
+message MsgLaunchConsumerChain {
+  // the title of this launch 
+  string title;
+  // the description of this consumer chain
+  string description;
+  // the chain-id of the new consumer chain
+  string chain_id;
+  // the proposed initial height of new consumer chain.
+  ibc.core.client.v1.Height initial_height;
+
+  ...  (all fields contained in ConsumerAdditionProposal except top_N)
+  
+  uint32 validators_power_cap;
+  uint32 validator_set_cap;
+  repeated string allowlist;
+  repeated string denylist;
+}
+```
+
+`MsgLaunchConsumerChain` contains everything that is contained in [`ConsumerAdditionProposal`](https://github.com/cosmos/interchain-security/blob/v4.3.0/proto/interchain_security/ccv/provider/v1/provider.proto#L29)
+except the `top_N` field because we can only launch Opt In consumer chains with this message (Top N chains still need to go
+through a governance proposal). Note, that an alternative format for this message could have been to include a single `ConsumerAdditionProposal` field in it,
+but this confuses the semantics of `MsgLaunchConsumerChain` and that of a governance proposal. The response of this message
+(i.e., `MsgLaunchConsumerChainResponse`) contains a single `string` that is the `consumerID`. The `consumerID`s corresponds
+to `len(chainID) | chainID | counter` and this way we can have multiple different `consumerID`s that correspond to a chain
+with the same `chainID`. By doing this, we tackle the main issue behind permissionless consumer chains, that is chain-id squatting.
+Note that `consumerID` could just be a `uint64` but we choose to include `chainID` in it as well so that it is easier to see
```

Comment:
I'm okay with having this simply as a `counter`. However, having an `uint64` would need substantially more refactoring because the currently used `chainId` is a `string`.  Also, having the `counter` stored as a `string` would simplify our life if later on we intend to introduce more data to the `consumerId`. Finally, note that `clientId` is similarly a `string`, although it could simply be a `uint64`.

Reply:
Not sure that I understand why is this more refactor. We need to migrate the state anyway to change the key. Does it matter how the new keys look like or what type we use? The keys are marshaled to []byte anyway.